File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    
    public String getKey(char[] arr) {
        StringBuilder sb = new StringBuilder("");
        for(char ch : arr) sb.append(ch);
        
        return sb.toString();
    }
    
    public List<List<String>> groupAnagrams(String[] strs) {
        int n = strs.length;
        HashMap<String, ArrayList<String>> map = new HashMap<>();
        for(int i = 0; i < n; i++) {
            char[]  arr = strs[i].toCharArray();
            Arrays.sort(arr);
            String key = getKey(arr);
            if(map.containsKey(key)) {
                map.get(key).add(strs[i]);
            } else {
                map.put(key, new ArrayList<String>());
                map.get(key).add(strs[i]);
            }
        }
        List<List<String>> ans = new ArrayList<List<String>>();
        for(String key : map.keySet()) {
            ans.add(map.get(key));
        }
        return ans;
    }
}

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    
    const map = new Map()
    const n = strs.length
    for(let i = 0 ; i < n; i++) {
        let key = strs[i].split("").sort().join('');
        if(map.has(key) == false) {
            map.set(key, []);
            map.get(key).push(strs[i]);
        } else {
            map.get(key).push(strs[i]);
        }
    }
    
    let answer = []
    for(let key of map.keys()) {
        answer.push(map.get(key))
    }
    return answer
};